keyserver/internal/device_list_update.go

```diff
@@ -367,6 +367,9 @@ func (u *DeviceListUpdater) processServer(serverName gomatrixserverlib.ServerNam
 					waitTime = fcerr.RetryAfter
 				} else if fcerr.Blacklisted {
 					waitTime = time.Hour * 8
+				} else {
+					// For all other errors (DNS resolution, network etc.) wait 1 hour.
+					waitTime = time.Hour
```

Comment:
How can this be hit at all when the error type is a `fedsenderapi.FederationClientError`?

Reply:
The `FederationClientError` has a `RetryAfter` of `0` and `Blacklisted` is `false`. So no path here is hit.
This should probably be fixed in the `FederationClient` later on, this is just a quick fix.